components/raftstore/src/store/snap.rs

```diff
@@ -493,36 +451,22 @@ impl Snap {
         Ok(s)
     }
 
-    pub fn new_for_building<T: Into<PathBuf>>(
+    fn new_for_building<T: Into<PathBuf>>(
         dir: T,
         key: &SnapKey,
-        size_track: Arc<AtomicU64>,
-        deleter: Box<dyn SnapshotDeleter>,
-        limiter: Limiter,
-        mgr: Option<Arc<DataKeyManager>>,
+        mgr: &SnapManagerCore,
     ) -> RaftStoreResult<Self> {
-        let mut s = Self::new(dir, key, size_track, true, true, deleter, limiter, mgr)?;
+        let mut s = Self::new(dir, key, true, true, mgr)?;
         s.init_for_building()?;
         Ok(s)
     }
 
-    pub fn new_for_sending<T: Into<PathBuf>>(
+    fn new_for_sending<T: Into<PathBuf>>(
         dir: T,
         key: &SnapKey,
-        size_track: Arc<AtomicU64>,
-        deleter: Box<dyn SnapshotDeleter>,
-        mgr: Option<Arc<DataKeyManager>>,
+        mgr: &SnapManagerCore,
     ) -> RaftStoreResult<Self> {
-        let mut s = Self::new(
-            dir,
-            key,
-            size_track,
-            true,
-            false,
-            deleter,
-            Limiter::new(INFINITY),
-            mgr,
-        )?;
+        let mut s = Self::new(dir, key, true, false, mgr)?;
```

Comment:
Is it a problem that original limit is INFINITY?

Reply:
I'm not sure why the original is INFINITY. I will keep the original value.